src/code/z_play.c

```diff
@@ -1,59 +1,1464 @@
+                        func_800C20B4(&globalCtx->preRenderCtx, &sp70);
+                        R_PAUSE_MENU_MODE = 2;
+                    } else {
+                        D_80161490 = 2;
+                    }
+                    gfxCtx->overlay.p = sp70;
+                    globalCtx->preRenderCtx.unk_A3 = 2;
+                    SREG(33) |= 1;
+                } else {
+                Gameplay_Draw_DrawOverlayElements:
+                    if ((HREG(80) != 10) || (HREG(89) != 0)) {
+                        Gameplay_DrawOverlayElements(globalCtx);
+                    }
+                }
+            }
+        }
+    }
+
+    if (globalCtx->view.unk_124 != 0) {
+        Vec3s sp50;
+        func_800591EC(&sp50, globalCtx->cameraPtrs[globalCtx->activeCamera]);
+        func_800AB944(&globalCtx->view);
+        globalCtx->view.unk_124 = 0;
+        if ((globalCtx->skyboxId != 0) && (globalCtx->skyboxId != 0x1D) && !globalCtx->envCtx.skyDisabled) {
+            func_800B1030(&globalCtx->skyboxCtx, globalCtx->view.eye.x, globalCtx->view.eye.y, globalCtx->view.eye.z);
+        }
+    }
+
+    func_80059EC8(globalCtx->cameraPtrs[globalCtx->activeCamera]);
+
+    Graph_CloseDisps(dispRefs, gfxCtx, "../z_play.c", 4508);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_play/Gameplay_Draw.s")
+#endif
+
+void Gameplay_Main(GlobalContext* globalCtx) {
+    D_8012D1F8 = &globalCtx->state.input[0];
+
+    DebugDisplay_Init();
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 4556);
+    }
+
+    if ((HREG(80) == 10) && (HREG(94) != 10)) {
+        HREG(81) = 1;
+        HREG(82) = 1;
+        HREG(83) = 1;
+        HREG(84) = 3;
+        HREG(85) = 1;
+        HREG(86) = 1;
+        HREG(87) = 1;
+        HREG(88) = 1;
+        HREG(89) = 1;
+        HREG(90) = 15;
+        HREG(91) = 1;
+        HREG(92) = 1;
+        HREG(93) = 1;
+        HREG(94) = 10;
+    }
+
+    if ((HREG(80) != 10) || (HREG(81) != 0)) {
+        Gameplay_Update(globalCtx);
+    }
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 4583)
+    }
+
+    Gameplay_Draw(globalCtx);
+
+    if (1 && HREG(63) != 0) {
+        LOG_NUM("1", 1, "../z_play.c", 4587);
+    }
+}
+
+// original name: "Game_play_demo_mode_check"
 s32 func_800BFC84(GlobalContext* globalCtx) {
-    return globalCtx->csCtx.state != 0 || func_8008E988(globalCtx) != 0;
+    return (globalCtx->csCtx.state != 0) || func_8008E988(globalCtx);
 }
 
+#ifdef NON_MATCHING
+// saved register usage differences and possibly stack usage or regalloc differences
+f32 func_800BFCB8(GlobalContext* globalCtx, MtxF* mf, Vec3f* vec) {
+    PosRot sp50;
+    f32 temp_f0;
+    f32 phi_f12;
+    f32 phi_f16;
+    f32 sp40;
+    f32 sp3C;
+    f32 sp38;
+    f32 sp34;
+
+    sp40 = func_8003CB30(globalCtx, &globalCtx->colCtx, &sp50, mf);
+
+    if (sp40 > -32000.0f) {
+        sp3C = sp50.rot.x * 0.00003051851f;
```

Comment:
0.0003051851 is 1.0f / SHRT_MAX, I will have a macro for this in an upcoming pr as well.  
```suggestion
        sp3C = sp50.rot.x * (1.0f / 32767.0f);
```